### ISSUE TITLE
Deprecated Hot Reload Docs

### ISSUE THREAD

**Issue by MostafaYehia** [nestjs/docs.nestjs.com#1554]
## Current behavior

 Error  Invalid options object. Watch Ignore Plugin has been initialized using an options object that does not match the API schema.
 - options[0] misses the property 'paths'. Should be:
   [RegExp | string, ...] (should not have fewer than 1 item)
   -> A list of RegExps or absolute paths to directories or files that should be ignored.
 - options[1] misses the property 'paths'. Should be:
   [RegExp | string, ...] (should not have fewer than 1 item)
   -> A list of RegExps or absolute paths to directories or files that should be ignored.

## Input Code

```ts
plugins: [
            ...options.plugins,
            new webpack.WatchIgnorePlugin([/\.js$/, /\.d\.ts$/]),
            ...
]
```

## Possible Solution

```ts
plugins: [
            ...options.plugins,
            new webpack.WatchIgnorePlugin({
                paths: [/\.js$/, /\.d\.ts$/]
            }),
            .
            .
],
```

## Environment
<pre><code>
Nest CLI version: 7.5.2 (Latest)

 
For Tooling issues:
- Node version: 12.14.0 
- Platform:  Windows 

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Would you like to create a PR for this issue?

**Comment by Tony133** [comment]
@kamilmysliwiec  I did this pull #1568  for this problem in the documentation

I hope it will go well :-)

**Comment by dadepra** [comment]
Ddid it was work for me! Thanks.

**Comment by PurpleEyeGH** [comment]
Not work for me.

**Comment by Tony133** [comment]
I write here the message I wrote in the pull, so I can clarify the problem as there is a bit of confusion

if you use nest 7.5, use webpack 5 and the solution is correct but you need to install
```start-server-nestjs-webpack-plugin``` instead of ```start-server-webpack-plugin````

Installation:

```$ npm i --save-dev webpack-node-externals start-server-nestj-webpack-plugin```

and create file webpack-hmr.config.js

```
const webpack = require('webpack');
const nodeExternals = require('webpack-node-externals');
const StartServerPlugin = require('start-server-nestjs-webpack-plugin');

module.exports = function(options) {
  return {
    ...options,
    entry: ['webpack/hot/poll?100', options.entry],
    externals: [
      nodeExternals({
        allowlist: ['webpack/hot/poll?100'],
      }),
    ],
    plugins: [
      ...options.plugins,
      new webpack.HotModuleReplacementPlugin(),
      new webpack.WatchIgnorePlugin({
          paths: [/\.js$/, /\.d\.ts$/]
      }),
      new StartServerPlugin({ name: options.output.filename }),
    ],
  };
};
```

In package.json change "npm start: dev" like so:

```"start:dev": "nest build --webpack --webpackPath webpack-hmr.config.js"```



I add a reference which may be helpful to this problem: ---> nestjs/nest#5706

**Comment by brandon-leapyear** [comment]
Copied from #5928

> So I actually got it working by replacing webpack-node-externals with webpack-pnp-externals. In summary, the following instructions get hot-reload to work with yarn 2 (modulo deprecation warnings):
>
> yarn add -D webpack-pnp-externals run-script-webpack-plugin webpack@5.9.0 (completely replaces the existing npm install instruction in the docs)
>
> Add the following webpack-hmr.config.js file:
```
const webpack = require('webpack');
const nodeExternals = require('webpack-node-externals');
const { RunScriptWebpackPlugin } = require('run-script-webpack-plugin');
const { WebpackPnpExternals } = require('webpack-pnp-externals');

module.exports = function(options) {
  return {
    ...options,
    entry: ['webpack/hot/poll?100', options.entry],
    watch: true,
    externals: [
      WebpackPnpExternals({ exclude: ['webpack/hot/poll?100'] }),
    ],
    plugins: [
      ...options.plugins,
      new webpack.HotModuleReplacementPlugin(),
      new webpack.WatchIgnorePlugin({
        paths: [/\.js$/, /\.d\.ts$/]
      }),
      new RunScriptWebpackPlugin({ name: options.output.filename }),
    ],
  };
};
```
> The differences with the webpack-hmr.config.js in the docs are:
* Use RunScriptWebpackPlugin instead of StartServerWebpackPlugin (StartServerWebpackPlugin incompatible with Webpack 5)
* Fix WatchIgnorePlugin to use paths key
* Use WebpackPnpExternals instead of webpack-node-externals (yarn 2 does not have a node_modules directory)
> Change start:dev script as usual
>
> In total, to resolve this issue, I'd recommend adding the above instructions to the NestJS docs about Hot Reload for Yarn v2, as well as fixing the DEP_WEBPACK_WATCH_WITHOUT_CALLBACK deprecation message (i believe this happens even without Yarn v2)

**Comment by m4r4c00ch0** [comment]
I followed [this](https://github.com/nestjs/docs.nestjs.com/issues/1554#issuecomment-740778901) instructions and I get the following error:

```
> nest build --webpack --webpackPath webpack-hmr.config.js

 Error  The 'compilation' argument must be an instance of Compilation
 ```

**Comment by gergelypap** [comment]
Just for the record, none of the suggestions mentioned above worked for me.

Using: yarn 1.22.10, node v14.15.4, @nestjs/core@7.6.11, webpack@5.9.0

**Comment by Tony133** [comment]
Hi @gergelypap,  If you have time, you can leave a minimal reproduction of a clonable git repository with the problem?

**Comment by gergelypap** [comment]
@Tony133  I am sorry. On a fresh install of Nest typescript starter, and applying your solution from https://github.com/nestjs/docs.nestjs.com/issues/1554#issuecomment-740778901, I confirm hot reloading works.

**Comment by kamilmysliwiec** [comment]
Let's track this here https://github.com/nestjs/docs.nestjs.com/pull/1621

**Comment by roboflank** [comment]
@m4r4c00ch0 Regarding the error ` Error  The 'compilation' argument must be an instance of Compilation`, I solved this by using `webpack@5.24`
